packages/ra-core/src/form/choices/useChoicesContext.ts

```diff
@@ -0,0 +1,48 @@
+import { useContext, useMemo } from 'react';
+import { RaRecord } from '../../types';
+import { useList } from '../../controller';
+import { ChoicesContext, ChoicesContextValue } from './ChoicesContext';
+
+export const useChoicesContext = <ChoicesType extends RaRecord = RaRecord>(
```

Comment:
There is something here that I don't understand: how does passing a `choices` prop override the 3 choices keys in the result? In other terms, how does it NOT break SelectInput and AutocompleteInput when used standalone?

Reply:
Because they only use the `allChoices`